djunicode/ergo
# Added Initialisation Function

## Description

> Initialisation Function is set up to get various information about the host system, such as if various programs are installed : node, npm python, etc , check for which platform it is running.

## Changes

- Added file ```initial.js``` at location ```app/electron/```
- Executed the function in ```app/electron/main.js```
- Dependencies Installed  :  ```shelljs``` , ```electron-store```

## Issue

> Closes issue :  #4

## Impacted Area

> The ```config.json``` file containing information during the initialisation of the electron app is stored at : 
  ```C:\Users\[USER_NAME]\AppData\Roaming\ergo```

## Steps to test

Steps needed to reproduce the scenario from this change to validate if the pull request solve this issue

- require the function from ```app/electron/initial.js``` into ```app/electron/main.js``` and then call the function inside ```main.js``` , run the app ```npm run dev```
- Check the ```config.json``` file stored at :  ```C:\Users\[USER_NAME]\AppData\Roaming\ergo``` 

## Screenshots

- Screenshot of the config.json file : 

> ![image](https://user-images.githubusercontent.com/68227858/107872592-52204980-6ed1-11eb-9b6d-43527fcc7558.png)
